src/vl-demo.js

```diff
@@ -0,0 +1,104 @@
+import { VlElement, define } from '/node_modules/vl-ui-core/dist/vl-core.js';
+import '/node_modules/prismjs/prism.js';
+import '/node_modules/prismjs/plugins/line-numbers/prism-line-numbers.js';
+import '/node_modules/prismjs/plugins/normalize-whitespace/prism-normalize-whitespace.js';
+import '/node_modules/vl-ui-titles/dist/vl-titles.js';
+
+/**
+ * VlDemo
+ * @class
+ * @classdesc 
+ * 
+ * @extends VlElement
+ * 
+ * @property {boolean} data-vl-title - Attribuut wordt gebruikt om de demo titel aan te geven.
+ * 
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-demo/releases/latest|Release notes}
+ * @see {@link https://www.github.com/milieuinfo/webcomponent-vl-demo/issues|Issues}
+ * @see {@link https://webcomponenten.omgeving.vlaanderen.be/demo/vl-demo.html|Demo}
+ * 
+ */
+export class VlDemo extends VlElement(HTMLElement) {
+    static get _observedAttributes() {
+        return ['title'];
+    }
+
+    constructor() {
+        super(`
+            <style>
+                @import '/node_modules/vl-ui-titles/src/style.css';
+                @import '/src/style.css';
+            </style>
+            <div>
+                <h3 is="vl-h3">Demo</h3>
+                <div class="demo">
+                    <slot></slot>
+                </div>
+                <pre class="line-numbers language-markup">
+                    <code></code>
+                </pre>
+            </div>
+        `);
+    }
+
+    get _titleElement() {
+        return this._shadow.querySelector('h3');
+    }
+
+    get _preElement() {
+        return this._shadow.querySelector('pre');
+    }
+
+    get _codeElement() {
+        return this._preElement.querySelector('code');
+    }
+
+    get _slotElement() {
+        return this._shadow.querySelector('slot');
+    }
+
+    connectedCallback() {
+        this._renderCode();
+    }
+
+    _renderCode() {
+        const elements = this._slotElement.assignedElements().map(element => this._getCode(element.cloneNode(true))).join('\n\t\t\t');
```

Comment:
is die join een poging om te "pretty printen"? is daar geen lib voor?

Reply:
Zonder join is dit het resultaat:

<img width="2880" alt="1" src="https://user-images.githubusercontent.com/11139249/82934609-0537c400-9f8c-11ea-9d45-3dd81cdcab76.png">

Met join, maar zonder 3 x tab is dit het resultaat:

<img width="2880" alt="2" src="https://user-images.githubusercontent.com/11139249/82934507-d7527f80-9f8b-11ea-971c-568b387f4494.png">

Het leek me bij ontwikkeling niet de moeite, maar ik zal nog eens onderzoeken of ik een alternatieve oplossing kan voorzien zonder het joinen met enter + 3 x tab.

Reply:
@pascalriquier ik heb de code kunnen wijzigen zodat er geen join met tabs meer nodig is.